### PR TITLE
[UT] fix asan UT failure & unstable UT

### PR DESCRIPTION
**Pull request by luohaha**
## What I'm doing:
fix asan UT failure & unstable UT

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [x] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.2
  - [x] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

File: be/test/storage/sstable/persistent_index_sstable_test.cpp [hunk 1]
```diff
@@ -102,6 +104,8 @@ TEST_F(PersistentIndexSstableTest, test_generate_sst_seek_and_check) {
         IndexValue cur_val(UNALIGNED_LOAD64(iter->value().get_data()));
         ASSERT_TRUE(exp_val == cur_val);
     }
+    delete iter;
+    delete sstable;
```
Comment (imay) [inline be/test/storage/sstable/persistent_index_sstable_test.cpp:108]: sstable should be initialized by nullptr.

Comment (luohaha) [reply be/test/storage/sstable/persistent_index_sstable_test.cpp:108]: Done

File: be/test/storage/lake/meta_file_test.cpp [hunk 2]
```diff
@@ -53,32 +53,32 @@ class TestLocationProvider : public LocationProvider {
 
 class MetaFileTest : public ::testing::Test {
 public:
-    static void SetUpTestCase() {
+    void SetUp() {
```
Comment (wyb) [inline be/test/storage/lake/meta_file_test.cpp:56]:
```suggestion
    void SetUp() override {
```

File: be/test/storage/lake/meta_file_test.cpp [hunk 3]
```diff
@@ -53,32 +53,32 @@ class TestLocationProvider : public LocationProvider {
...
-                std::make_unique<lake::TabletManager>(s_location_provider.get(), s_update_manager.get(), 1638400000);
+        _location_provider = std::make_unique<FixedLocationProvider>(kTestDir);
+        _mem_tracker = std::make_unique<MemTracker>(1024 * 1024);
+        _update_manager = std::make_unique<lake::UpdateManager>(_location_provider.get(), _mem_tracker.get());
+        _tablet_manager =
+                std::make_unique<lake::TabletManager>(_location_provider.get(), _update_manager.get(), 1638400000);
     }
 
-    static void TearDownTestCase() { (void)FileSystem::Default()->delete_dir_recursive(kTestDir); }
+    void TearDown() { (void)FileSystem::Default()->delete_dir_recursive(kTestDir); }
```
Comment (wyb) [inline be/test/storage/lake/meta_file_test.cpp:68]:
```suggestion
    void TearDown() override { (void)FileSystem::Default()->delete_dir_recursive(kTestDir); }
```

**Review by wyb — APPROVED** [review]

**Review by sevev — APPROVED** [review]

**Comment by luohaha** [comment]
https://github.com/Mergifyio backport branch-3.2

**Comment by luohaha** [comment]
https://github.com/Mergifyio backport branch-3.1

**Pull request merged**